src/Battlescape/BattlescapeState.cpp

```diff
@@ -1372,6 +1374,29 @@ void BattlescapeState::blinkVisibleUnitButtons()
 }
 
 /**
+ * Shifts the colors of the health bar when unit has fatal wounds.
+ */
+void BattlescapeState::blinkHealthBar()
+{
+	static Uint8 color = _barHealth->getColor(), maxcolor = color + 3, step = 0;
+
+	BattleUnit *bu = _save->getSelectedUnit();
+	if (bu == 0 || !_barHealth->getVisible()) return;
+
+	for (int i = 0; i < 6; i++)
+	{
+		if (bu->getFatalWound(i) > 0)
+		{
+			_barHealth->setColor(color);
+			color += step ^= 1;		// +1, +0, +1, +0, ...
```

Comment:
Possible speedup of flashing if this line will be change to
```
color++;
```
May be will be even better, but this depends on the taste.